c2rust-analyze/src/c_void_casts.rs

```diff
@@ -274,16 +287,88 @@ impl<'tcx> CVoidCasts<'tcx> {
     /// [`Cast`]: rustc_middle::mir::Rvalue::Cast
     /// [`From`]: CVoidCastDirection::From
     /// [`To`]: CVoidCastDirection::To
-    pub fn should_skip_stmt(&self, stmt: &Statement<'tcx>) -> bool {
-        let (lhs, rv) = match get_assign_sides(stmt) {
-            None => return false,
-            Some(it) => it,
-        };
-        self.to.contains(lhs)
-            || self.from.contains(match get_cast_place(rv) {
-                None => return false,
-                Some(it) => it,
-            })
+    pub fn should_skip_stmt(&self, loc: &Location) -> bool {
+        self.to.contains(loc) || self.from.contains(loc)
+    }
+
+    fn is_place_local_modified(p: &Place, stmt: &Statement) -> bool {
+        if !p.projection.is_empty() {
+            return false;
+        }
+
+        let local = p.local;
+
+        if let StatementKind::Assign(asgn) = stmt.kind.clone() {
+            let (lhs, rv) = *asgn;
+            if lhs.local == local {
+                return true;
+            }
+
+            use Rvalue::*;
+            let rv_place = match rv {
+                Use(op) => op.place(),
+                Repeat(op, _) => op.place(),
+                Ref(_, _, p) => Some(p),
+                // ThreadLocalRef
+                AddressOf(_, p) => Some(p),
+                Len(p) => Some(p),
+                Cast(_, op, _) => op.place(),
+                // BinaryOp
+                // CheckedBinaryOp
+                // NullaryOp
+                UnaryOp(_, op) => op.place(),
+                Discriminant(p) => Some(p),
+                // Aggregate
+                ShallowInitBox(op, _) => op.place(),
+                _ => None,
+            };
```

Comment:
```suggestion
        if let StatementKind::Assign(assign) = &stmt.kind {
            let (lhs, rv) = &**assign;
            if lhs.local == local {
                return true;
            }

            use Rvalue::*;
            let rv_place = match rv {
                Use(op) => op.place(),
                Repeat(op, _) => op.place(),
                Ref(_, _, p) => Some(*p),
                // ThreadLocalRef
                AddressOf(_, p) => Some(*p),
                Len(p) => Some(*p),
                Cast(_, op, _) => op.place(),
                // BinaryOp
                // CheckedBinaryOp
                // NullaryOp
                UnaryOp(_, op) => op.place(),
                Discriminant(p) => Some(*p),
                // Aggregate
                ShallowInitBox(op, _) => op.place(),
                _ => None,
            };
```

Don't need to clone `StatementKind` and its `Box`.

Reply:
4bb545f4